javascript/javascript1/week3/note_taking_app.js

```diff
@@ -0,0 +1,62 @@
+// SAVE A NOTE
+const notepad = [];
+function saveNote(notes, content, id) {
+    const draft = {};
+    draft["content"] = content;
+    draft.id = id;
+    notes.push(draft);
+    
+}
+saveNote(notepad, "Pick up groceries", 1);
+saveNote(notepad, "Do laundry", 2);
+console.log(notepad);
+
+
+// GET A NOTE 
+function getNote (notes, id) {
+    for (let i = 0; i<notes.length; i++) {
+        if (notes[i].id === id) {
+            return notes[i];
+        }
+    }
+    return "ERROR!";
+}
+
+const firstNote = getNote(notepad, 1);
+console.log(firstNote);
+
+const thirdNote = getNote(notepad, 3);
+console.log(thirdNote);
+
+
+// LOG OUT A NOTE 
+function logOutNotesFormatted(notes) {
+    for (let i = 0; i<notes.length; i++) {
+        const idString = notes[i].id;
+        const contentString = notes[i].content;
+        console.log(`The note with id: ${idString}, has the following note text: ${contentString}`)
+    }
+}
+
+logOutNotesFormatted(notepad);
+
+
+//UNIQUE FEATURE: DELETING A NOTE
+function noteDelete (notes, idOrContent) {
+    for (let i = 0; i < notes.length; i++) {
+        if (notes[i].id === idOrContent || notes[i]["content"].includes(idOrContent)) {
```

Comment:
When you are searching for a string within a string it is easier when you use `.toLowerCase()` on both sides so it is case insensitive.
```suggestion
        const contentInNote = notes[i].content.toLowerCase();
        if (notes[i].id === idOrContent || contentInNote.includes(idOrContent.toLowerCase())) {
```

But this is only possible if the `idOrContent` is a string!

Reply:
This is very helpful, thanks! Should I have used ` .trim( )` to be space key insensitive, too?